src/modules/85-iacm/components/IACMPipelineSteps/IACMApproval/StepMode.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ReactElement } from 'react'
+import { Formik, FormikForm, FormInput } from '@harness/uicore'
+import * as Yup from 'yup'
+import type { FormikProps } from 'formik'
+
+import {
+  FormMultiTypeDurationField,
+  getDurationValidationSchema
+} from '@common/components/MultiTypeDuration/MultiTypeDuration'
+import { setFormikRef, StepFormikFowardRef } from '@pipeline/components/AbstractSteps/Step'
+import { useVariablesExpression } from '@pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression'
+import { getNameAndIdentifierSchema } from '@pipeline/components/PipelineSteps/Steps/StepsValidateUtils'
+import { useStrings } from 'framework/strings'
+
+import type { IACMApprovalData, IACMApprovalStepProps } from './types'
+
+const IACMApprovalStepMode = (
+  props: IACMApprovalStepProps,
+  ref: StepFormikFowardRef<IACMApprovalData>
+): ReactElement => {
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+
+  const { initialValues, allowableTypes, stepViewType, readonly, isNewStep, onUpdate } = props
+
+  const handleSubmit = (value: IACMApprovalData): void => {
+    if (onUpdate) {
+      onUpdate(value)
+    }
+  }
+
+  const validationSchema = Yup.object().shape({
```

Comment:
Not an issue, just an FYI... you can combine `object` and `shape`. Makes zero difference to how it works, so either way is fine 👍🏻
```suggestion
  const validationSchema = Yup.object({
```